metricbeat/module/zookeeper/mntr/data.go

```diff
@@ -53,7 +53,8 @@ var (
 		"approximate_data_size": c.Int("zk_approximate_data_size"),
 	}
 	schemaLeader = s.Schema{
-		"followers":        c.Int("zk_followers"),
+		"learners":         c.Int("zk_learners", s.Optional),
+		"followers":        c.Int("zk_followers", s.Optional), // Not present anymore in ZooKeeper >= 3.6 mntr responses
```

Comment:
Umm, if this is a field that has been renamed, from followers to learners, we can keep both fields, and eventually remove the old one.

There was a similar case in redis, where some fields were renamed in 5.0. On #8167 we [renamed](https://github.com/elastic/beats/pull/8167/files#diff-55387d78bdbcd48ce8f65e10c3c2d4bce39e8c406b33212ccff10b9caecdf35eR83) the fields, storing the values obtained from each one of the versions on both fields. Eventually we removed the old field in https://github.com/elastic/beats/pull/28246, keeping compatibility with older versions by copying the old metric into the new field.

The advantage of this is that dashboards or other queries can work the same, independently of the Zookeeper version.

Wdyt about following a similar approach here?

(This could be done in a follow-up in any case)

Reply:
Yes, exactly that case! So just to make sure I understood you proposal: The idea is to add a new "ingested" field which represents both source fields?

The current approach in this PR is: Add the new source field as a new ingested field, keep them both for compatibility purposes.

We would need to change the dashboards either way.

Or is it maybe that we would have two ingested fields "followers" (old) and "learners" (new) fed with the same whatever source field is present?

Reply:
The idea would be to expect the same fields in the documents, abstracting the Zookeeper version, so dashboard or queries work the same.

> Or is it maybe that we would have two ingested fields "followers" (old) and "learners" (new) fed with the same whatever source field is present?

Yes, it would be like that, to have the two fields always ingested. On old versions with the value of "zk_followers", and in new versions with the value of "zk_learners". We would deprecate "followers" and eventually remove it, but we could continue filling "learners" with the value of "zk_followers" in old versions.

Another approach could be to continue using the "followers" field, with the value of "zk_learners" in newer versions, but it'd be probably better to migrate to the new terminology used by Zookeeper.

Any of these changes would require more code after applying `schemaLeader`.

Reply:
Thanks for the suggestion! In 97e5518a6d8409e2ad4c706e6c97e4d1d89d5a23 I added some logic to fill the "learners" field with "followers" (modern view of legacy ZK versions) or to fill the "followers" field with "learners" (back compatibility with old dashboards when using ZK>3.6).

@jsoriano The ITs only run ensembles with one ZK node, meaning that they are always "standalone" which implies that the fields "zk_followers" or "zk_learners" will not ever be present in the tested mntr outputs. I've tried to find, with no success, an unit test for the fields extraction. How do you unit test this code?